hs_communities/views/communities.py

```diff
@@ -0,0 +1,138 @@
+            'groups': groups,
+            'grpfilter': grpfilter,
+        }
+
+
+class FindCommunitiesView(TemplateView):
+    template_name = 'hs_communities/find-communities.html'
+
+    def dispatch(self, *args, **kwargs):
+        return super(FindCommunitiesView, self).dispatch(*args, **kwargs)
+
+    def get_context_data(self, **kwargs):
+
+        return {
+            'communities_list': Community.objects.all()
+        }
+
+
+@method_decorator(login_required, name='dispatch')
+class MyCommunitiesView(TemplateView):
+    template_name = 'hs_communities/my-communities.html'
+
+    def dispatch(self, *args, **kwargs):
+        return super(MyCommunitiesView, self).dispatch(*args, **kwargs)
+
+    def group_to_community(self, grp, communities):
+        """
+        return the community membership information of a group; group can belong to only one community
+        :param grp: group object
+        :param communities: communities
+        :return: tuple id, name of community
+        """
+        for community in communities:
+            if grp.id in [g.id for g in community.member_groups]:
+                return community
+
+    def get_context_data(self, **kwargs):
+        grps_member_of = []
+        u = User.objects.get(pk=self.request.user.id)
+        groups = Group.objects.filter(gaccess__active=True).exclude(name="Hydroshare Author")
+        # for each group set group dynamic attributes
+        for g in groups:
+            g.is_user_member = u in g.gaccess.members
+            if g.is_user_member:
+                grps_member_of.append(g)
+            g.join_request_waiting_owner_action = g.gaccess.group_membership_requests.filter(request_from=u).exists()
+            g.join_request_waiting_user_action = g.gaccess.group_membership_requests.filter(invitation_to=u).exists()
+            g.join_request = None
+            if g.join_request_waiting_owner_action or g.join_request_waiting_user_action:
+                g.join_request = g.gaccess.group_membership_requests.filter(request_from=u).first() or \
+                                 g.gaccess.group_membership_requests.filter(invitation_to=u).first()
+
+        comms_member_of = [self.group_to_community(g, Community.objects.all()) for g in grps_member_of]
+        return {
+            'communities_list': [c for c in comms_member_of if c is not None]
+        }
+
+
+@method_decorator(login_required, name='dispatch')
+class TopicsView(TemplateView):
+    """
+    action: CREATE, READ, UPDATE, DELETE
+    """
+
+    def get(self, request, *args, **kwargs):
+        return render(request, 'pages/topics.html', {'topics_json': self.get_topics_data()})
+
+    def post(self, request, *args, **kwargs):
+        if request.POST.get('action') == 'CREATE':
+            try:
+                new_topic = Topic()
+                new_topic.name = request.POST.get('name').replace("--", "")
+                new_topic.save()
+            except Exception as e:
```

Comment:
Could you be more specific with these exceptions you want to catch?

Reply:
I didn't hit any Exceptions yet, was trying to avoid a 500 for unforeseen. I'd rather just log here and keep the broad exception.

Reply:
Without proper monitoring of the logs (which we do not have), broad exception catches are likely to go unnoticed when they occur.  I really think we need to be more specific with the exceptions we catch and handle with only a few exceptions that may occur on a case by case basis.  Adding a new topic is not a good exception to the rule.

Reply:
As this is the narrow use-case and we don't have that proper mointoring in place, I'm inclined to have it go without an Exception, then. An admin should know when topic creation fails for any reason, as it would likely be some unforeseen infrastructure issue.

Reply:
Addressed in 5d9396d30